src/main/java/stef/projects/console/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,170 @@
+        statement.setString(1, email);
+        ResultSet resultSet = statement.executeQuery();
+        User user = null;
+        while (resultSet.next()) {
+            showUser(resultSet);
+            user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+        }
+        return user;
+    }
+
+    @Override
+    public User findUsernameAndPassword(String username, String password) throws SQLException { //find or validate
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SELECT_BY_USERNAME_AND_PASSWORD_QUERY);
+        statement.setString(1, username);
+        statement.setString(2, password);
+        ResultSet resultSet = statement.executeQuery();
+        User user = null;
+        while (resultSet.next()) {
+            showUser(resultSet);
+            user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+        }
+        return user;
+    }
+
+    @Override
+    public List<User> findUsersByUserRole(UserRole userRole) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SELECT_BY_USER_ROLE_QUERY);
+        statement.setString(1, UserRole.getValue(userRole));
+        ResultSet resultSet = statement.executeQuery();
+        List<User> userList = new ArrayList<>();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            User user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            userList.add(user);
+        }
+        return userList;
+    }
+
+    @Override
+    public List<User> findUserByFirstNameOrLastName(String fistName, String lastName) throws SQLException {
+        PreparedStatement statement = DatabaseConnection.getPreparedStatementFromQuery(SELECT_BY_FIRST_OR_LAST_NAME_QUERY);
+        statement.setString(1, fistName);
+        statement.setString(2, lastName);
+        ResultSet resultSet = statement.executeQuery();
+        List<User> userList = new ArrayList<>();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            User user = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            userList.add(user);
+        }
+        return userList;
+    }
+
+    private boolean extractStatus(int status) {
```

Comment:
1) Else branch is not needed actually
if (status == 1) { 
  return true;
}
return false;
2) Maybe makes sense to move it on the DatabaseConnection class? will be used by other repositories as well